### DISCUSSION TITLE
Error: failed to create new program: No such project

### DISCUSSION THREAD

**Discussion by thomasschus** [deta/cloud-docs#729] [Help] [answered]
Following these [steps](https://fastapi.tiangolo.com/deployment/deta/) on Linux Mint 21.1 Vera gives me the below error when trying to deploy my application with the Deta CLI:

**Error: failed to create new program: No such project**

I've already tried specifying name, project and runtime - without success. It feels like if Deta CLI does not detect my main.py.

What can I do?

**Comment by jnsougata** [comment, marked as answer]
do you have `default` project?
also can you add the commands you are trying?

**Reply by thomasschus** [reply]
That was it. I've missed creating the default project. I've simply not seen the step of creating a default project in this [deployment guide](https://fastapi.tiangolo.com/deployment/deta/). Cheers!

**Comment by MasterpieceElbow** [comment]
@jnsougata 
Hi! 
Can you explain what the '`default` project' means?
I have the same issue as @thomasschus but I haven't found any information about having a `default` project.

**Reply by jnsougata** [reply]
If you are a new user visit https://web.deta.sh/ and create a new project named `default` or it will prompt you to create a default project under which your bases, drives and micros will be created if project is not mentioned with a CLI command

**Reply by MasterpieceElbow** [reply]
@jnsougata Thanks a lot!
Solution: you have to `deta new --python first_micro` for creating the first micro for python, it was mentioned almost nowhere.

**Reply by jnsougata** [reply]
> @jnsougata Thanks a lot!
> 
> Solution: you have to `deta new --python first_micro` for creating the first micro for python, it was mentioned almost nowhere.
> 
> 

The `first_micro` is name of the micro you are creating and it's optional... if it's not given it will use root dir name as the name of the micro. The issue is due to the fact that your deta account didn't have a default project so CLI was unable to put the micro under the default project as you have not created it in deta cloud dashboard or you didn't specify the target project name with the CLI command.
